src/view/components/semantic/elements/input-date/demo.ts

```diff
@@ -1,8 +1,68 @@
 import {bootstrap} from 'angular2/bootstrap'
 import {Attribute, Component, View} from 'angular2/core'
 import {CORE_DIRECTIVES, FORM_DIRECTIVES} from 'angular2/common'
+import {InputDate} from './input-date'
 
-import {InputDate, InputDateModel} from './input-date'
+
+export class InputDateModel {
+  name:string
+  placeholder:string
+  value:string
+  disabled:boolean
+  icon:string
+  type:string
+
+  constructor(name:string = null,
+              placeholder:string = '',
+              type:string = 'date',
+              value:string = null,
+              disabled:boolean = null,
+              icon:string = '') {
+
+    this.name = !!name ? name : 'field-' + new Date().getTime() + Math.floor(Math.random() * 1000)
+    this.placeholder = placeholder
+    this.type = type
+    this.value = value
+    this.disabled = disabled
+    this.icon = icon || ''
+    if(this.icon.indexOf(' ') == -1 && this.icon.length > 0){
+      this.icon = (this.icon + ' icon').trim()
+    }
+  }
+
+  validateDate(date:string) {
```

Comment:
eeeek!

Can you take a look at momentjs and take a guess at how long you think it would take you to add it for use here?  http://momentjs.com/